xdsl/transforms/print_to_println.py

```diff
@@ -0,0 +1,180 @@
+from dataclasses import dataclass
+from typing import Iterable, Sequence
+import hashlib
+import re
+
+from xdsl.ir import SSAValue, Attribute, MLContext, Operation
+from xdsl.dialects import print, builtin, arith, llvm
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    RewritePattern,
+    op_type_rewrite_pattern,
+    PatternRewriteWalker,
+)
+
+from xdsl.passes import ModulePass
+
+
+i8 = builtin.IntegerType(8)
+
+
+def legalize_str(val: str):
+    """
+    Takes any string and legalizes it to be a global llvm symbol.
+    (for the strictes possible interpreation of this)
+
+     - Replaces all whitespaces and dots with _
+     - Deletes all non ascii alphanumerical characters
+
+    The resulting string consists only of ascii letters, underscores and digits
+    """
+    val = re.sub(r"(\s+|\.)", "_", val)
+    val = re.sub(r"[^A-Za-z0-9_]+", "", val).strip("_")
+    return val
+
+
+def _key_from_str(val: str) -> str:
+    """
+    Generate a symbol name form any given string.
+
+    Takes the first ten letters of the string plus it's sha1 hash to create a
+    (pretty much) globally unique symbol name.
+    """
+    h = hashlib.new("sha1")
+    h.update(val.encode())
+    return f"{legalize_str(val[:10])}_{h.hexdigest()}"
```

Comment:
I'm wondering if there could be scenario's where we want exactly similar e.g. format strings to live in different parts of memory, they would be represented with the same symbol here, right? Would we then need to represent them with a different symbol?

Reply:
I don't think so, actually